cpp/CMakeLists.txt

```diff
@@ -788,7 +804,12 @@ target_compile_definitions(cudf PUBLIC SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${RMM_LO
 # - link libraries --------------------------------------------------------------------------------
 
 # link targets for cuDF
-target_link_libraries(cudf arrow arrow_cuda nvrtc ${CUDART_LIBRARY} cuda ${ZLIB_LIBRARIES} ${Boost_LIBRARIES})
+target_link_libraries(cudf 
+                      arrow arrow_cuda 
+                      nvrtc ${CUDART_LIBRARY} 
+                      ZLIB::ZLIB Boost::filesystem 
+                      rmm dlpack 
+                      ${GTEST_LIBRARY} ${benchmark_LIBRARY})
```

Comment:
the cudf target shouldn't need gtest or benchmark, only the tests need those

Reply:
OK.

Reply:
To make this work with a pre-installed RMM, as opposed to one pulled in and used via `add_subdirectory()`, you should link to `rmm::rmm`, which will work for both cases.